docs/docs-beta/docs/dagster-plus/deployment/deployment-types/multi-tenancy.md

```diff
@@ -1,5 +1,138 @@
+         # ...
+   ```
+
+---
+
+## Isolating execution context between projects
+
+Separating execution context between projects can have several motivations:
+
+- Facilitating separation of duty between teams to prevent access to sensitive data
+- Differing compute environments and requirements, such as different architecture, cloud provider, etc.
+- Reducing impact on other projects. For example, a project with a large number of runs can impact the performance of other projects.
+
+In order from least to most isolated, there are three levels of isolation:
+
+- [Code location](#code-location-isolation)
+- [Agent](#agent-isolation)
+- [Deployment](#deployment-isolation)
+
+### Code location isolation
+
+If you have no specific requirements for isolation beyond the ability to deploy and run multiple projects, you can use a single agent and deployment to manage all your projects as individual code locations.
+
+![Diagram of isolation at the code location level](/images/dagster-cloud/managing-deployments/isolation-level-code-locations.png)
+
+| **Pros** | **Cons** |
+|----------|----------|
+| <ul><li>Simplest and most cost-effective solution</li><li>User access control can be set at the code location level</li><li>Single glass pane to view all assets</li></ul> | No isolation between execution environments |
+
+### Agent isolation
+
+:::note
+Agent queues are only available on [Hybrid deployment](/dagster-plus/deployment/deployment-types/hybrid/).
+:::
+
+{/* Using the [agent routing feature](/dagster-plus/deployment/agents/running-multiple-agents#routing-requests-to-specific-agents), you can effectively isolate execution environments between projects by using a separate agent for each project. */}
+Using the [agent routing feature](/todo), you can effectively isolate execution environments between projects by using a separate agent for each project.
+
+Motivations for utilizing this approach could include:
+
+- Different compute requirements, such as different cloud providers or architectures
+- Optimizing for locality or access, such as running the data processing closer or in environment with access to the storage locations
+
+![Diagram of isolation at the agent level](/images/dagster-cloud/managing-deployments/isolation-level-agents.png)
+
+| **Pros** | **Cons** |
+|----------|----------|
+| <ul><li>Isolation between execution environments</li><li>User access control can be set at the code location level</li><li>Single glass pane to view all assets</li></ul> | Extra work to set up additional agents and agent queues |
+
+### Deployment isolation
+
+:::note
+Multiple deployments are only available in Dagster+ Pro.
+:::
+
+Of the approaches outlined in this guide, multiple deployments are the most isolated solution. The typical motivation for this isolation level is to separate production and non-production environments. It may be considered to satisfy other organization specific requirements.
```

Comment:
Feel like the last sentence doesn't really add anything.

Reply:
agree